big_tests/tests/sasl_external_SUITE.erl

```diff
@@ -107,22 +127,39 @@ generate_user(C, User, TemplateValuesIn) ->
     Cmd2 = [SignCmd, "--req", UserCsr, "--out", UserCert],
     LogFile = filename:join(?config(priv_dir, C), User ++ "signing.log"),
     {done, 0, _} = erlsh:run(Cmd2, LogFile, SSLDir),
+    #{key => UserKey,
+      cert => UserCert}.
+
+
+generate_user_tcp(C, User) ->
+    generate_user(C, User, escalus_tcp).
+
+generate_user(C, User, Transport) ->
+    Certs = ?config(certs, C),
+    UserCert = maps:get(User, Certs),
+
+    Common = [{username, list_to_binary(User)},
+	      {server, <<"localhost">>},
+	      {password, <<"break_me">>},
+	      {resource, <<>>}, %% Allow the server to generate the resource
+	      {auth, {escalus_auth, auth_sasl_external}},
+	      {transport, Transport},
+	      {ssl_opts, [{certfile, maps:get(cert, UserCert)},
+			  {keyfile, maps:get(key, UserCert)}]}],
+    Common ++ transport_specific_options(Transport).
 
-    [{username, list_to_binary(User)},
-     {server, <<"localhost">>},
-     {password, <<"break_me">>},
-     {resource, <<>>}, %% Allow the server to generate the resource
-     {auth, {escalus_auth, auth_sasl_external}},
-     {ssl_opts, [{certfile, UserCert},
-		 {keyfile, UserKey}]},
-     {starttls, required}].
+transport_specific_options(escalus_tcp) ->
+    [{starttls, required}];
+transport_specific_options(_) ->
+     [{port, 5285},
```

Comment:
Is there a way not to "hard code" port number?

Reply:
Or Is there a reason to do so?

Reply:
You have to pass the port number in some way anyway. The port is already specified in test.config only reading it from there would make the test case more complex to reason about in my opinion.

Reply:
How about a "common_const" module in which you have functions returning consts?
eg 
```erlang
-module(sasl_external_SUITE_consts).
-export([transport_specific_port/0]).

transport_specific_port() -> 5285.
```
This might make looking for port number easier, because you use function name instead of a number.

Reply:
This sounds like a good idea, probably worth its own PR :)